gui/RespiraConnectedDevice.h

```diff
@@ -0,0 +1,115 @@
+#include "../common/generated_libs/network_protocol/network_protocol.pb.h"
+#include "../common/third_party/nanopb/pb_common.h"
+#include "../common/third_party/nanopb/pb_decode.h"
+#include "../common/third_party/nanopb/pb_encode.h"
+#include "connected_device.h"
+#include <QSerialPort>
+#include <QThread>
+
+#define INTER_FRAME_TIMEOUT 11
+
+class RespiraConnectedDevice : public ConnectedDevice {
+
+public:
+  RespiraConnectedDevice(QString portName, uint32_t write_timeout_ms,
+                         uint32_t interframe_timeout_ms)
+      : serialPortName(portName), m_writeTimeout_ms(write_timeout_ms),
+        m_interframeTimeout_ms(interframe_timeout_ms) {}
+
+  ~RespiraConnectedDevice() {
+    if (m_serial != nullptr) {
+      m_serial->close();
+    }
+  }
+  void createPort() {
+    if (m_serial != nullptr)
+      return;
+    printf("Socket create\n");
+    m_serial = new QSerialPort();
+    m_serial->setPortName(serialPortName);
+    m_serial->setBaudRate(QSerialPort::Baud115200);
```

Comment:
Please add comments explaining why these specific parameter values were chosen. Are all of them necessary btw?

Reply:
These parameters fully specify the serial port configuration we use, some of them are default (like parity, data bits, stop bits) but I think it's good to be explicit here.